doc/example-ast-destructing.mld

```diff
@@ -0,0 +1,171 @@
+{0 Destructuring AST}
+
+{1:table-of-contents Table of Contents}
+
+- {{!section-description} Description}
+- {{!section-"ast-structure-pattern-matching"} AST Structure Pattern Matching}
+{ul {- {{!section-"example-matching-integer-payload-manually"} Example: Matching Integer Payload Manually}}}
+
+- {{!section-"using-ast_pattern-high-level-destructors"} Using {!Ppxlib.Ast_pattern} High-Level Destructors}
+{ul {- {{!section-"example-1-matching-integer-payload-with-ast_pattern"} Example 1: Matching Integer Payload with [Ppxlib.Ast_pattern]}}}
+{ul {- {{!section-"example-2-simplifying-matching-with-eint"} Example 2: Simplifying Matching with [eint]}}}
+
+- {{!section-"using-metaquot"} Using Metaquot}
+{ul {- {{!section-"example-1-matching-integer-payload-with-metaquot"} Example 1: Matching Integer Payload with Metaquot}}}
+{ul {- {{!section-"example-2-matching-complex-expressions-with-metaquot-and-anti-quotations"} Example 2: Matching Complex Expressions with Metaquot and Anti-Quotations}}}
+
+- {{!section-"conclusion"} Conclusion}
+
+{1:description Description}
+
+Destructuring an AST (Abstract Syntax Tree) is essential when creating a PPX (preprocessor extension) in OCaml. To generate or transform code, you must first break down the AST to understand and manipulate its structure.
+
+For example, if you want to transform this code:
+
+{[
+let one = [%one]
+]}
+
+into:
+
+{[
+let one = 1
+]}
+
+You’ll need to destructure the AST representing the extension point ([%one]) to replace it with [1].
+There are several ways to destructure an AST. We’ll explore three methods:
+
+- {b AST Structure Pattern Matching}
+- {b Using {!Ppxlib.Ast_pattern} High-Level Destructors}
+- {b Using Metaquot}
+
+{1:ast-structure-pattern-matching AST Structure Pattern Matching}
+
+The most fundamental method for destructuring an AST in PPXLib is by directly matching on the AST’s structure.
```

Comment:
Looking through the current documentation and README, it seems to me that we tend to write Ppxlib rather than PPXLib. Perhaps we could even change all occurrences of PPXLib into `{! Ppxlib}`.